### PR TITLE
New grad template for takeovers

### PR DESCRIPTION
**Pull request by pmahnke**
## Done

- created a template and scss for the new takeover templates

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8001/takeovers/templates
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Compare to the design [here](https://github.com/canonical-web-and-design/web-squad/issues/1556)

## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/1610

## Screenshots

large
image

small
image

### CODE REVIEW

File: templates/takeovers/_template.html [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+      {% if subtitle %}<h4>
+        {{ subtitle }}
+      </h4>{% endif %}
+      <div class="u-hide--small">
+        <p>
+          <a href="{{ url }}" class="{% if cta_class %}{{ cta_class }}{% else %}p-button--positive{% endif %}">{% if url == "http" %}<span class="p-link--external">{% endif %}{{ cta }}{% if url == "http" %}</span>{% endif %}
+          </a>
+          <p>
+        <p>
+          <a href="{{ url2 }}" {% if url3 == "http" %}class="p-link--external"{% endif %}>
```
Comment (sowasred2012) [inline templates/takeovers/_template.html]: Should this if statement be `{% if 'http' in url2 %}`? Doesn't look like `url3` is passed in in the example.

File: templates/takeovers/_template.html [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
+      <img src="{% if 'http' not in image %}https://assets.ubuntu.com/v1/{% endif %}{{ image }}" alt="" {% if image_style %}style="{{ image_style }}"{% endif %}>
+    </div>
+    <div class="u-hide--medium u-hide--large">
+      <p>
+        <a href="{{ url }}" class="{% if cta_class %}{{ cta_class }}{% else %}p-button--positive{% endif %}">{% if 'http' not in url %}<span class="p-link--external">{% endif %}{{ cta }}{% if 'http'  in url %}</span>{% endif %}
+        </a>
+      <p>
+      {% if url2 %}
+      <p>
+        <a href="{{ url2 }}" {% if 'http' not in url2 %}class="p-link--external"{% endif %}>
```
Comment (sowasred2012) [inline templates/takeovers/_template.html]: Similar question to above - if `http` were not present, wouldn't that suggest a relative path i.e. not external?

Comment (pmahnke) [reply templates/takeovers/_template.html]: good catch, fixed

File: templates/takeovers/_template.html (hunk 1 above)
Comment (pmahnke) [reply templates/takeovers/_template.html]: good catch, fixed

**Comment by anasereijo** [comment]
@pmahnke image on the desktop version doesn't seem to be centred

**Comment by anasereijo** [comment]
much better @pmahnke!!! yay

File: static/sass/_pattern_takeovers.scss [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+//}
+
+@mixin ubuntu-p-takeovers {
+  .p-takeover {
+    @extend %vf-strip;
+    background-blend-mode: multiply, multiply, normal, normal;
+    background-color: $color-x-light;
+    background-image:
+    linear-gradient(
+    to bottom left,
```
Comment (anthonydillon) [inline static/sass/_pattern_takeovers.scss]: Under indented

File: static/sass/_pattern_takeovers.scss [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    linear-gradient(
+    -89deg,
+    #e95420 0%,
+    #772953 42%,
+    #2c001e 94%
+    );
+    background-position: top right, top left, right 100%, left top;
+    background-repeat: no-repeat;
+    background-size: 74% 99.83%, 68% 91%, 102.5% 18%, 100% 99.8%;
+    color: $color-light;
```
Comment (anthonydillon) [inline static/sass/_pattern_takeovers.scss]: Shouldn't that be `$color-x-light`?

File: static/sass/_pattern_takeovers.scss [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    p:last-of-type {
+      margin-bottom: 0 !important;
+    }
+
+    .col-6,
+    .col-7 {
+      margin-bottom: 1rem;
+    }
+
+    &__rightcol {
```
Comment (anthonydillon) [inline static/sass/_pattern_takeovers.scss]: I would suggest creating a new utility called `u-align--left-medium` which applies the left on medium and smaller.

File: static/sass/_pattern_takeovers.scss [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+      padding-top: 0;
+    }
+
+    p:last-of-type {
+      margin-bottom: 0 !important;
+    }
+
+    .col-6,
+    .col-7 {
+      margin-bottom: 1rem;
```
Comment (anthonydillon) [inline static/sass/_pattern_takeovers.scss]: Do we need this? Shouldn't the strip padding give you the space you want?

File: templates/takeovers/_template.html [hunk 7]
```diff
@@ -1,14 +1,41 @@
...
-
-<div class="row equal-height">
-	<div class="six-col box box-highlight">
-		
-	</div><!-- /.six-col -->
-	<div class="six-col last-col box box-highlight">
-		
-	</div><!-- /.six-col -->
-</div><!-- / .row -->
+<section lang="{% if lang %}{{ lang }}{% else %}en{% endif %}" lang="fr" data-lang="{% if locale %}{{ locale }}{% else %}en_GB{% endif %}" class="{% if class %}{{ class }}{% else %}p-takeover{% endif %} js-takeover">
```
Comment (anthonydillon) [inline templates/takeovers/_template.html]: You have two lang attr

File: templates/takeovers/_template.html [hunk 8]
```diff
@@ -1,14 +1,41 @@
...
+    <div class="{% if equal_cols %}col-6{% else %}col-7{% endif %}">
+      <h1>
+        {{ title }}
+      </h1>
+      {% if subtitle %}<h4>
+        {{ subtitle }}
+      </h4>{% endif %}
+      <div class="u-hide--small">
+        <p>
+          <a href="{{ url }}" class="{% if cta_class %}{{ cta_class }}{% else %}p-button--positive{% endif %}">{% if 'http' in url %}<span class="p-link--external">{% endif %}{{ cta }}{% if 'http' in url %}</span>{% endif %}
```
Comment (anthonydillon) [inline templates/takeovers/_template.html]: Should you check there is a `url` just in case

File: templates/takeovers/_template.html [hunk 9]
```diff
@@ -1,14 +1,41 @@
...
+          </a>
+        </p>
+        {% if url2 %}
+        <p>
+          <a href="{{ url2 }}" {% if 'http' in url2 %}class="p-link--external"{% endif %}>
+            {{ cta2 }}
```
Comment (anthonydillon) [inline templates/takeovers/_template.html]: If not external we should add `&nbsp;&rsaquo;`

File: templates/takeovers/_template.html [hunk 10]
```diff
@@ -1,14 +1,41 @@
...
+          </a>
+        </p>
+        {% endif %}
+      </div>
+    </div>
+    <div class="{% if equal_cols %}col-6 {% else %}col-5{% endif %} u-vertically-center p-takeover__rightcol">
```
Comment (anthonydillon) [inline templates/takeovers/_template.html]: Don't need the space after col-6

File: templates/takeovers/_template.html [hunk 11]
```diff
@@ -1,14 +1,41 @@
...
+      <img src="{% if 'http' not in image %}https://assets.ubuntu.com/v1/{% endif %}{{ image }}" alt="" {% if image_style %}style="{{ image_style }}"{% endif %}>
```
Comment (anthonydillon) [inline templates/takeovers/_template.html]: You should check there is an `image`

File: templates/takeovers/_template.html [hunk 12]
```diff
@@ -1,14 +1,41 @@
...
+    </div>
+    <div class="u-hide--medium u-hide--large">
+      <p>
+        <a href="{{ url }}" class="{% if cta_class %}{{ cta_class }}{% else %}p-button--positive{% endif %}">{% if 'http' in url %}<span class="p-link--external">{% endif %}{{ cta }}{% if 'http'  in url %}</span>{% endif %}
```
Comment (anthonydillon) [inline templates/takeovers/_template.html]: Check url here too

File: templates/takeovers/_template.html [hunk 13]
```diff
@@ -1,14 +1,41 @@
...
+        </a>
+      </p>
+      {% if url2 %}
+      <p>
+        <a href="{{ url2 }}" {% if 'http' in url2 %}class="p-link--external"{% endif %}>
+          {{ cta2 }}
```
Comment (anthonydillon) [inline templates/takeovers/_template.html]: Same comment about the >

File: static/sass/_pattern_takeovers.scss (hunk 4 above)
Comment (pmahnke) [reply static/sass/_pattern_takeovers.scss]: is it?

File: templates/takeovers/_template.html [hunk 14]
```diff
@@ -1,14 +1,47 @@
...
+        {% endif %}
+      </div>
+    </div>
+    {% if image %}
+    <div class="{% if equal_cols %}col-6{% else %}col-5{% endif %} {% if image_hide_small is sameas true %}u-hide--small{% endif %} u-vertically-center u-align--center u-align--left-medium">
+      <img src="{% if 'http' not in image %}https://assets.ubuntu.com/v1/{% endif %}{{ image }}" alt="" {% if image_style %}style="{{ image_style }}"{% endif %}>
+    </div>
+    {% endif %}
+    <div class="u-hide--medium u-hide--large">
+      {% if url %}
```
[the remark this answers is not in the document]
Comment (pmahnke) [reply templates/takeovers/_template.html]: done.

File: static/sass/_pattern_takeovers.scss [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+//@include ubuntu-p-takeover-neutral;
+//@include ubuntu-p-takeover-dark;
+//@include ubuntu-p-takeover-snapcraft;/
+//}
+
+@mixin ubuntu-p-takeovers {
+  .p-takeover {
+    @extend %vf-strip;
+    background-blend-mode: multiply, multiply, normal, normal;
+    background-color: $color-x-light;
```
[the remark this answers is not in the document]
Comment (pmahnke) [reply static/sass/_pattern_takeovers.scss]: Done.

**Pull request merged**